hazelcast/src/main/java/com/hazelcast/nio/serialization/GenericRecordBuilder.java

```diff
@@ -110,6 +110,8 @@ static GenericRecordBuilder compact(String typeName) {
     /**
      * It is legal to set the field again only when Builder is created with {@link GenericRecord#cloneWithBuilder()}.
      * Otherwise, it is illegal to set to the same field twice.
+     * <p>
+     * Supported only for {@link Portable}. Not applicable for {@link com.hazelcast.config.CompactSerializationConfig Compact}
```

Comment:
We need to put a similar comment to `GenericRecordBuilder setArrayOfChar(@Nonnull String fieldName, @Nullable char[] value);`